vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/server/config_test.go

```diff
@@ -93,9 +97,14 @@ func TestNewWithDelegate(t *testing.T) {
 	wrappingServer.PrepareRun()
 	wrappingServer.RunPostStartHooks(stopCh)
 
+
 	server := httptest.NewServer(wrappingServer.Handler)
 	defer server.Close()
 
+	// Wait for the hooks to finish before checking the response
+	<-delegatePostStartHookChan
```

Comment:
what is the goal here? I think this only postpones the race a bit. the race is still there. Instead check that `/healthz` gives you 200.

Reply:
hmm, those are tested below. What exactly is seen in the race?

Reply:
For the lines
```
[+]poststarthook/delegate-post-start-hook ok
[+]poststarthook/wrapping-post-start-hook ok
```
this fix seems to be right. Do we see in test runs that the `/healthz` check fails? I don't see it in the logs of https://github.com/openshift/origin/issues/21875.

Reply:
@sttts https://github.com/openshift/origin/issues/21875

The delegate hook report failed. I was able to reproduce it in slow container in 3/10 runs. When I added sleep there it worked fine, so I figured this happens when go routines are not scheduled fast enough and the post-hook is still pending or haven't run yet. This fix will make sure both hooks ran before we go check the result.

Reply:
After slack discussion, lgtm (issue log were incomplete).

Reply:
@sttts updated issue